### DISCUSSION TITLE
Restricted instances

### DISCUSSION THREAD

**Discussion by bijlevel** [RomanKubiak/ctrlr#309] [Q&A]
Does anyone know if creating restricted instances is a JUCE 'feature' or a Roman Kubiak 'feature'? The latest build v5.6.23 does export an instance WITHOUT the menubar, which makes it of course useless. I'm trying to find out if I can 'repair' this myself, but have no clue where to look for it in the code.

**Comment by RomanKubiak** [comment]
It's my feature i'll hvae a look what's broken and why it's not showing the menubar (i'm sure it's a property of a panel if the menubar should be visible or not)

**Comment by bijlevel** [comment]
Thanks! And could you also have a look why the panel isn't resizable when exported as (restricted) instance?

**Comment by Godlike-Productions** [comment]
@RomanKubiak If you resize in the VST version of  Ctrlr, the background doesn't redraw. You need to close and reopen the  VST.  Perhaps something missing in the resize callback or repaint.  I'm not sure if it's related to the menu, but both issues were fixed for the 6.0.26 version in Windows.

**Reply by bijlevel** [reply]
For the last days I'm trying to find out why the menubar is ALWAYS hidden when exporting a (restricted) instance. Roman is right, it is a property of the panel one can set. It is part of a set of properties that influence the menubar when exporting an instance:

<img width="556" alt="Screenshot 2021-05-30 at 11 12 43" src="https://user-images.githubusercontent.com/11859363/120098779-1899e400-c138-11eb-8cb5-dc87179d6450.png">

But because the first property (hide/show menubar on export) does not work, one can't test the other properties! I did inspect meticulous the code related to these properties, but it seems that part is OK. Somehow now I have a feeling that MAYBE it has to do with the menubar height. Is it possible that the height is set to zero when exporting?

@RomanKubiak could you PLEASE try to find time to solve this issue, because it is keeping us from publishing working plugins?
